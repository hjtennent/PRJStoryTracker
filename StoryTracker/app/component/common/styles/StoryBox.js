import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  similarStoryBox: {
    flexDirection: 'row',
    flex: 1,
    borderWidth: 1,
    borderColor: 'lightgrey',
    marginBottom: 20,
    alignItems: 'center',
    padding: 10,
  },
  headlineContainer: {
    flex: 0.8,
  },
  similarityLabelContainer: {
    flex: 0.2,
    justifyContent: 'center',
    alignItems: 'center'
  },
  storyText: {
    fontFamily: 'Oxygen-Regular'
  }
});

export default styles;