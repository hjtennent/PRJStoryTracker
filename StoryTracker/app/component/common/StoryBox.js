import React from 'react';
import styles from "./styles/StoryBox";
import {
  View, TouchableOpacity, Text
} from 'react-native';

export default StoryBox = (props) => {

  return (
    <View style={styles.similarStoryBox}>
      <View style={styles.headlineContainer}>
        <TouchableOpacity onPress={() => props.onPress(props.story[1]['url'])}>
          <Text style={styles.storyText}>{props.story[0]}</Text>
        </TouchableOpacity>
      </View>
      <View style={styles.similarityLabelContainer}>
        <Text style={styles.storyText}>{props.story[1]['score']}</Text>
      </View>
    </View>
  )
}