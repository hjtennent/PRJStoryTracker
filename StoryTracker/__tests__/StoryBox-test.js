import React from 'react';
import renderer from 'react-test-renderer';
import StoryBox from '../app/component/common/StoryBox';
import { configure } from 'enzyme';
import { render, fireEvent } from '@testing-library/react-native';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('StoryBox renders correctly', () => {
  const tree = renderer.create(<StoryBox onPress={jest.fn()} text={"www.test.com"} 
                                  url={"www.test.com"} similarity="0.0"/>
                              ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('StoryBox onPress function fires', () => {
  const mockFn = jest.fn();
  const { getByText } = render(
    <StoryBox onPress={mockFn} text={"www.test.com"} url="www.test.com"
      similarity="0.0" />
  );
  fireEvent.press((getByText('www.test.com')))
  expect(mockFn.mock.calls.length).toBe(1);
})