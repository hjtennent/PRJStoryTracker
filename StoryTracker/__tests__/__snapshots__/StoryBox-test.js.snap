// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryBox renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "lightgrey",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 20,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.8,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Oxygen-Regular",
          }
        }
      >
        www.test.com
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Oxygen-Regular",
        }
      }
    >
      0.0
    </Text>
  </View>
</View>
`;
